pkgs/by-name/op/opentelemetry-cpp/package.nix

```diff
@@ -39,12 +44,18 @@ stdenv.mkDerivation (finalAttrs: {
 
   buildInputs = [
     curl
-    grpc
     nlohmann_json
-    prometheus-cpp
-    protobuf
   ];
 
+  propagatedBuildInputs =
+    lib.optionals enableGrpc [
+      protobuf
```

Comment:
protobuf is not only required by the grpc exporter, but also by the http exporter. The condition for `lib.optionals` therefore should be `(enableGrpc || enableHttp)`.